src/main/java/dev/seniorjava/githubstatisticsbot/stats/mappers/StatisticsDeserializer.java

```diff
@@ -0,0 +1,49 @@
+package dev.seniorjava.githubstatisticsbot.stats.mappers;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import dev.seniorjava.githubstatisticsbot.stats.collectors.ReactionStatistics;
+import dev.seniorjava.githubstatisticsbot.stats.collectors.ReactionStatistics.Reactions;
+import dev.seniorjava.githubstatisticsbot.stats.collectors.ReactionStatistics.User;
+import java.io.IOException;
+
+public class StatisticsDeserializer extends JsonDeserializer<ReactionStatistics> {
+
+  @Override
+  public ReactionStatistics deserialize(final JsonParser parser, final DeserializationContext context)
+      throws IOException {
+
+    final JsonNode node = parser.getCodec().readTree(parser);
+    Reactions reactions = parseReactions(node.get("reactions"));
+    User user = parseUser(node.get("user"));
+
+    return new ReactionStatistics(user, reactions);
+  }
+
+  private Reactions parseReactions(final JsonNode reactions) {
+    if (reactions == null) {
+      return null;
+    }
+    return new Reactions(
+        reactions.get("+1").asInt(),
+        reactions.get("-1").asInt(),
+        reactions.get("laugh").asInt(),
+        reactions.get("hooray").asInt(),
+        reactions.get("confused").asInt(),
+        reactions.get("heart").asInt(),
+        reactions.get("rocket").asInt(),
+        reactions.get("eyes").asInt()
+    );
+  }
+
+  private User parseUser(final JsonNode user) {
+    if (user == null) {
+      return null;
+    }
```

Comment:
Optional  можно использовать, его для таких целей и создавали

Reply:
The `user` field is a part of the [ReactionStatistics](https://github.com/driversti/GithubStatisticsBot/pull/1/files/c94f68e356bd880a49c3e43319b01d785711067c#diff-9d5d5d151dab8ee46d004982d600cba487f0bb7d11480977ae2ac0662f696833) class structure. Do you find it convenient and correct to have fields in a class inside the `Optional` wrapper?

Reply:
я еще раз пересмотрел логику ,
насчет optinal филдов ,которые будет юзаться в классе ReactionStatistics  думаю это не очень подойдет.

просто сейчас выходит что ,может быть ситуация при которой вы создадите  
`  return new ReactionStatistics(null, null);`
это выглядит как-то странно да   и для этого вы пишите проверки на верхнем уровне и возвращайте 0 при null.

я бы предложил при null   - возвращать объкт с пред. установленными значениями , для Reactions  -> все филды будут заполенны 0 , и тогда не нужны будут в вызывающем коде проверки и как-по мне смотреться по приятнее

Reply:
Yes, you're right. We can get a case like this
```java
return new ReactionStatistics(null, null);
```
But the target solution may differ.
**One way:** we assume this is a normal case (as I did) and live it as is. As an additional improvement, we may add @Nullable annotation under the fields.
**Second way:** during mapping, we may return an object with initialized fields by default. But I'd like to avoid this because the mapper should not care about this. Its responsibility is mapping. That's all.
**Some other ways...**

Do you call `null` checks in the [ReactionStatistics](http://github.com/driversti/GithubStatisticsBot/blob/222e8a666e8aa959f066441ce77e90a45452bc16/src/main/java/dev/seniorjava/githubstatisticsbot/stats/collectors/ReactionStatistics.java) a _calling code_? If so I don't agree. I incapsulated that checks inside the owner. I'd like to avoid making such checks in the real _calling code_ [StatisticsFormatter](http://github.com/driversti/GithubStatisticsBot/blob/222e8a666e8aa959f066441ce77e90a45452bc16/src/main/java/dev/seniorjava/githubstatisticsbot/stats/formatters/StatisticsFormatter.java).